test/asyncprogressqueueworker.cc

```diff
@@ -0,0 +1,90 @@
+#include "napi.h"
+
+#include <chrono>
+#include <condition_variable>
+#include <mutex>
+#include <thread>
+
+#if (NAPI_VERSION > 3)
+
+using namespace Napi;
+
+namespace {
+
+struct ProgressData {
+  int32_t progress;
+};
+
+class TestWorker : public AsyncProgressQueueWorker<ProgressData> {
+public:
+  static Napi::Value CreateWork(const CallbackInfo& info) {
+    int32_t times = info[0].As<Number>().Int32Value();
+    Function cb = info[1].As<Function>();
+    Function progress = info[2].As<Function>();
+
+    TestWorker* worker = new TestWorker(cb, progress, "TestResource", Object::New(info.Env()));
+    worker->_times = times;
+
+    return Napi::External<TestWorker>::New(info.Env(), worker);
+  }
+
+  static Napi::Value QueueWork(const CallbackInfo& info) {
+    auto wrap = info[0].As<Napi::External<TestWorker>>();
+    auto worker = wrap.Data();
+    worker->Queue();
+    return Napi::Boolean::New(info.Env(), true);
+  }
+
+  static Napi::Value CancelWork(const CallbackInfo& info) {
```

Comment:
This function can also be void. No need to return a value.

Reply:
OTOH, if you wish to test that cancelling work works, then `Queue()` and/or `Cancel()` must return a value which will be passed through to JS here.